### PR TITLE
avoid duplicate get container

### PR DESCRIPTION
**Pull request by effi-ofer**
This change takes care of the double GET Container REST calls we have observed in tpc-ds.

Tested by running read 50, read 500, copy 50, wordcount, teragen, terasort, tpc-ds.  The read 500 workload in particular has close to 4000 objects, so we use it to stress the multiple pages logic.
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java [hunk 1]
```diff
@@ -483,17 +482,20 @@ public FSDataInputStream getObject(String hostName, Path path) throws IOExceptio
...
+    while (moreData) {
+      Collection<StoredObject> res = cObj.list(obj, marker, pageListSize);
+      moreData = (res.size() == pageListSize);
+      if (marker == null && (res == null || res.isEmpty() || res.size() == 0)) {
         FileStatus[] emptyRes = {};
         LOG.debug("List {} in container {} is empty", obj, container);
         return emptyRes;
       }
       for (StoredObject tmp : res) {
+        marker = tmp.getAsObject().getName();
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java]: @effi-ofer what this does? It just update the new value in a loop? We need this to get the latest name to use it as a marker for the next read? - If so, may be we can do something like
this marker = res.get(res.size).getName() ?

Comment (effi-ofer) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java]: Good suggestion Gil.  

The while loop spins while there is moreData.  Each time there is moreData it reads from the list using the prior marker.  We therefore need to set the marker each time we go through the while loop for the next read.  The marker is just the name of the last object that was returned.  We can pull it out of the for loop and do something exactly like you suggested.  I can do that, test it, and get back to you.

Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java]: @effi-ofer I am afraid a bit that doing tmp.getAsObject() in a loop will be costly a bit.

**Comment by gilv** [comment]
Thanks.

**Pull request closed without merging**